streams/src/main/java/org/apache/kafka/streams/AutoOffsetReset.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams;
+
+import java.time.Duration;
+import java.util.Optional;
+
+/**
+ * Sets the {@code auto.offset.reset} configuration when
+ * {@link #addSource(AutoOffsetReset, String, String...) adding a source processor} or when creating {@link KStream}
+ * or {@link KTable} via {@link StreamsBuilder}.
+ */
+public class AutoOffsetReset {
```

Comment:
How about renaming this to `StreamAutoOffsetReset`, as we now have `AutoOffsetReset` for consumer, share group, and stream?

- https://github.com/apache/kafka/pull/17858
- https://github.com/apache/kafka/pull/18096

Additionally, I believe we should rename the `AutoOffsetReset` class added in https://github.com/apache/kafka/pull/17858 for better clarity.  
@omkreddy, your thoughts?

Reply:
I'm also good with the name, it makes sense to me as it is an internal class.